### ISSUE TITLE
Enable locking slots in GT machines

### ISSUE THREAD

**Issue by ghost** [GregTechCEu/GregTechCEu-1.19#2626]
### Cross-mod Integration

Pipez and other I/O mods

### Feature Description

TL;DR: Introduce a feature to "remember" or "forget" the contents of specific inventory slots in blocks, similar to the functionality in RFTools, to prevent a single item type from filling an entire block's inventory.

Automating certain recipes in blocks like the MV Circuit Assembler becomes impractical due to the lack of slot control. For example, crafting Integrated Processors without the Large Circuit Assembling Facility is problematic when using external item transportation mods like Pipez. Pipez can transfer up to 1 stack per tick with maximum upgrades, which often results in one item type dominating the block's inventory. This behavior makes it impossible to automate crafting in blocks that rely on multiple distinct input types unless slower item pipes are used.
    
My proposed solution is to add two buttons next to the slots in a GregTech block's GUI:

**"Remember" Button:** Locks all slots to only accept the item currently in each slot at the time the button is pressed.
Ensures specific items always go to the correct slots, preventing inventory overflow by a single item type.
**"Forget" Button:** Resets all slots, allowing it to accept any item again.

This would make it a lot easier to automate processors and other complex items early-on in the game, since I don't like to live in a GUI manually making items. I prefer to stock my machines and do something else while my machines process.

Thanks for reading this feature request. I hope you add it soon.

**Comment by screret** [comment]
This not existing is intended, Use a robot arm on keep exact.

**Comment by ghost** [comment]
It's not possible to do this on recipes with more than 4 components, since there are only 6 faces on any GT machine and 2 of those are reserved for power and the output.

**Comment by MatthiasMann** [comment]
Use an arm with an item filter in keep exact mode => easy

**Comment by ghost** [comment]
I have no idea how to configure the robot arm. There are no tooltips or guides in any cover UIs, in fact. If you won't add slot locking consider making covers easier to understand and use.

If you don't believe me, boot up ATM 9 and use a robot arm. No pointers in any direction. At all.

**Comment by MatthiasMann** [comment]
the arm is easy to use (install it on the machine):
- top right is the mode button - it has a tooltip - set it to "keep exact"
- top left is the input/output direction - set it to input
- in the middle is the item filter once you inserted one (shift click on or place it in the slot on the right)
- when moving items from JEI/EMI and the mode is set to exact it will also set the amount from JEI/EMI
- bottom right is a switch which controls if other item movement can bypass the filter (then the filter only applies to items moved by the arm) or not (that's the default)
you can easily automate any single block machine with 2 AE2 interface and 1 arm and 1 pump plus optional level emitter and machine control cover on a 3rd site

**Comment by Spicierspace153** [comment]
![image](https://github.com/user-attachments/assets/10bc7e69-c6d5-40b7-9ad2-0b3e2680eefa)
![image](https://github.com/user-attachments/assets/12e86173-aaff-4e2c-b883-fe8165177378)
![image](https://github.com/user-attachments/assets/7dfc6b96-1ea8-467f-a8f0-aa5595b7bc35)
![image](https://github.com/user-attachments/assets/3abbe4c6-8d7e-4886-8eaf-7b253bbb8690)
This was a fresh install of atm9

**Comment by ghost** [comment]
So from what I can see this is set to import 6 stacks/s and keep 1 of each item in each slot (??). Is that right?

**Comment by Deepacat** [comment]
You can keep however much you want in each slot with a filter.

**Comment by MatthiasMann** [comment]
as I said several time: "keep exact" not "supply exact".
You use supply exact if you want to transfer a fixed amount of items (eg useful for the electrolyzer)
You use keep exact if you want to keep a fixed amount of items in the machine - eg for an assembler.

If you insert an item filter into the arm (as I wrote above) you can specify the amount per item in the filter (up to 9 items).

**Comment by Deepacat** [comment]
https://www.youtube.com/watch?v=0R6pWBJPD1g
Here I made a video example :P

**Comment by ghost** [comment]
Is there a way to do this without AE2, rather using GT's item pipes or Pipez? Or is AE2 required? I'm not very invested into the mod but if it's required I guess I'll use it.

**Comment by Deepacat** [comment]
Right now you cannot robot arm through gregtech pipes I believe, so you'd need something else to bring items/buffer. Also this mod is nigh unplayable without AE2 so I wouldn't recommend getting into it without learning AE2

**Comment by ghost** [comment]
Oh well. I guess it's time to make some cables and ME drives. Thanks for the help!

**Comment by ghost** [comment]
Also, one more thing. How come however I configure the robot arm, the ME Export Bus seems to only import items in quantities of 96?

**Comment by MatthiasMann** [comment]
the export bus pushes items - you need to use an interface so that the arm can pull
